07_control_statements_branching_and_jumps/exercises/order.c

```diff
@@ -0,0 +1,98 @@
+// of beets and later enters 5 pounds of beets, the program should use report 
+// 9 pounds of beets. The program then should compute the total charges, the 
+// discount, if any, the shipping charges, and the grand total. The program 
+// then should display all the purchase information: the cost per pound, the 
+// pounds ordered, and the cost for that order for each vegetable, the total 
+// cost of the order, the discount (if there is one), the shipping charge, and 
+// the grand total of all the charges.
+#include <stdio.h>
+
+#define ARTICHOKE 2.05
+#define BEET 1.15
+#define CARROT 1.09
+#define DISCOUNT 0.05
+#define OVER 100
+#define RATE1 6.5
+#define RATE2 14.0
+#define RATE3 0.5
+#define BREAK1 5
+#define BREAK2 20
+
+int main (void)
+{
+    float pound;
+    float cost_a = 0;
+    float cost_b = 0;
+    float cost_c = 0;
+    float pound_a = 0;
+    float pound_b = 0;
+    float pound_c = 0;
+    float discount = 0;
+    float all_cost;
+    float all_pound;
+    float delivery;
+    char category;
+
+    printf("Choose your category to enter the pounds (q to quit):\n");
+    printf("a) Artichoke $2.05\n");
+    printf("b) Beet      $1.15\n");
+    printf("c) Carrot    $1.09\n");  
+    while ((category = getchar()) != 'q')
+    {
+        switch (category)
+        {
+            case 'a' : printf("Enter weight in pounds:\n");
+                       scanf("%f", &pound);
+                       pound_a += pound;
+                       cost_a = pound_a * ARTICHOKE;
+                       break;
+            case 'b' : printf("Enter weight in pounds:\n");
+                       scanf("%f", &pound);
+                       pound_b += pound;
+                       cost_b = pound_b * BEET;
+                       break;
+            case 'c' : printf("Enter weight in pounds:\n");
+                       scanf("%f", &pound);
+                       pound_c += pound;
+                       cost_c = pound_c * CARROT;
+                       break;
+            default :  printf("Choose the number of your category or quit:\n");
+                       continue;
+        }
+    }
+        all_cost = cost_a + cost_b + cost_c;
+        all_pound = pound_a + pound_b + pound_c;
+        if (all_cost >= OVER)
+            discount = all_cost * DISCOUNT;
+        if (all_pound < BREAK1)
+            delivery = RATE1;
+        else if (all_pound <= BREAK2)
+            delivery = RATE2;
+        else if (all_pound > BREAK2)
+            delivery = RATE2 + (all_pound - BREAK2) * RATE3;
+        printf("Item        Amount      Price      Total\n");
+        printf("---------------------------------------------\n");
+        printf("Artichokes  %-10.2f  $%.2f      $%.2f\n", pound_a, ARTICHOKE, cost_a);
+        printf("Beets       %-10.2f  $%.2f      $%.2f\n", pound_b, BEET, cost_b);
+        printf("Carrots     %-10.2f  $%.2f      $%.2f\n", pound_c, CARROT, cost_c);
+        printf("---------------------------------------------\n");
```

Comment:
If a user order for a particular vegetable is less than 1 cent don't add it to the order. If total price is less than 1 cent print "Order is cancelled. Bye!"